### PR TITLE
Multistage builds for cn.ubuntu.com

### PR DESCRIPTION
**Pull request by carkod**
## Done

- [x] Based on ubuntu.com and jp.ubuntu.com
- [x] Copies over global-nav.js

## QA

- Build the image: `docker build --tag myimage --build-arg BUILD_ID=myfakeid .`
- Once image is built, `docker run -ti -p 8111:80 myimage`
- Check site is running smoothly at http://localhost:8111
- Check new id `myfakeid` is in the headers `curl -I http://localhost:8111`

## Issue

Fixes #2309

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -81,7 +81,6 @@ run
 .*.hash
 
 # Node is only needed for building, not running
-package.json
 node_modules
 yarn.lock
```
Comment (jpmartinspt) [inline .dockerignore]: Can you remove yarn.lock too?

Comment (carkod) [reply .dockerignore]: Done and added LICENSE and rebased.

File: Dockerfile [hunk 2]
```diff
@@ -1,23 +1,52 @@
...
+# ===
+FROM ubuntu:bionic AS python-dependencies
+RUN apt-get update && apt-get install --no-install-recommends --yes python3-pip python3-setuptools
+ADD requirements.txt /tmp/requirements.txt
+RUN --mount=type=cache,target=/root/.cache/pip pip3 install --user --requirement /tmp/requirements.txt
+
+
+# Build stage: Install yarn dependencies
+# ===
+FROM node:10-slim AS yarn-dependencies
```
Comment (tbille) [inline Dockerfile]: Could you make the update to node 12. Which is the latest LTS from node. And the one that we are using in the run script.

File: Dockerfile [hunk 3]
```diff
@@ -1,23 +1,52 @@
...
+
 # Set up environment
 ENV LANG C.UTF-8
 WORKDIR /srv
 
-# System dependencies
-RUN apt-get update && apt-get install -y --no-install-recommends python3 python3-setuptools python3-pip
+# Import code, build assets and mirror list
+ADD . .
+RUN rm -rf package.json yarn.lock .babelrc webpack.config.js
```
Comment (tbille) [inline Dockerfile]: Could you remove also the file: `requirements.txt`?

**Pull request merged**